### ISSUE TITLE
system-config-printer: No GSettings schemas are installed on the system

### ISSUE THREAD

**Issue by deepfire** [NixOS/nixpkgs#24880]
## Issue description
`system-config-printer` crashes, after clicking to add a `.ppd` file during new printer specification:
```
deepfire@andromedae:~$ system-config-printer
/nix/store/f7is1rsyz2mb0l0nc2q70az5qxaj3w7x-system-config-printer-1.5.7/share/system-config-printer/system-config-printer.py:36: PyGIWarning: GdkPixbuf was imported without specifying a version first. Use gi.require_version('GdkPixbuf', '2.0') before import to ensure that the right version gets loaded.
  from gi.repository import GdkPixbuf
/nix/store/f7is1rsyz2mb0l0nc2q70az5qxaj3w7x-system-config-printer-1.5.7/share/system-config-printer/system-config-printer.py:38: PyGIWarning: Gdk was imported without specifying a version first. Use gi.require_version('Gdk', '3.0') before import to ensure that the right version gets loaded.
  from gi.repository import Gdk

** (system-config-printer.py:1472): WARNING **: Error retrieving accessibility bus address: org.freedesktop.DBus.Error.ServiceUnknown: The name org.a11y.Bus was not provided by any .service files
/nix/store/f7is1rsyz2mb0l0nc2q70az5qxaj3w7x-system-config-printer-1.5.7/share/system-config-printer/system-config-printer.py:39: PyGIWarning: Gtk was imported without specifying a version first. Use gi.require_version('Gtk', '3.0') before import to ensure that the right version gets loaded.
  from gi.repository import Gtk
/nix/store/f7is1rsyz2mb0l0nc2q70az5qxaj3w7x-system-config-printer-1.5.7/share/system-config-printer/jobviewer.py:28: PyGIWarning: Notify was imported without specifying a version first. Use gi.require_version('Notify', '0.7') before import to ensure that the right version gets loaded.
  from gi.repository import Notify

(system-config-printer.py:1472): Gtk-WARNING **: Error loading theme icon 'edit-find' for stock: Icon 'edit-find' not present in theme GNOME

(system-config-printer.py:1472): Gtk-WARNING **: Error loading theme icon 'edit-clear' for stock:

(system-config-printer.py:1472): Gtk-WARNING **: Could not load a pixbuf from /org/gtk/libgtk/theme/Adwaita/assets/bullet-symbolic.svg.
This may indicate that pixbuf loaders or the mime database could not be found.
No ID match for device ipp://printer:631/ipp:
MFG:Generic;MDL:Printer;DES:Generic Printer;
Using drv:///cupsfilters.drv/textonly.ppd (status: 3)

(system-config-printer.py:1472): GLib-GIO-ERROR **: No GSettings schemas are installed on the system
Trace/breakpoint trap
```

### Steps to reproduce
- `nix-env -i system-config-printer`
- `system-config-printer`
- find a generic, network IPP printer, click to specify a .ppd file
- KABOOM

## Technical details

* System: (NixOS: `nixos-version`, Ubuntu/Fedora: `lsb_release -a`, ...)
* Nix version: (run `nix-env --version`)
* Nixpkgs version: (run `nix-instantiate --eval '<nixpkgs>' -A lib.nixpkgsVersion`)

**Comment by deepfire** [comment]
This is very related to https://github.com/NixOS/nixpkgs/issues/21647

**Comment by deepfire** [comment]
`strace` gives following:
```
deepfire@andromedae:~$ strace -f system-config-printer 2>scf.strace
No ID match for device ipp://printer:631/ipp:
MFG:Generic;MDL:Printer;DES:Generic Printer;
Using drv:///cupsfilters.drv/textonly.ppd (status: 3)

### ..try to add printer, and make it crash, then:

Trace/breakpoint trap
deepfire@andromedae:~$ ag compiled scf.strace
32309:[pid  4424] open("/run/current-system/sw/share/glib-2.0/schemas/gschemas.compiled", O_RDONLY) = -1 ENOENT (No such file or directory)
32310:[pid  4424] open("/nix/var/nix/profiles/default/share/glib-2.0/schemas/gschemas.compiled", O_RDONLY) = -1 ENOENT (No such file or directory)
32311:[pid  4424] open("/home/deepfire/.nix-profile/share/glib-2.0/schemas/gschemas.compiled", O_RDONLY) = -1 ENOENT (No such file or directory)
32312:[pid  4424] open("/run/opengl-driver/share/glib-2.0/schemas/gschemas.compiled", O_RDONLY) = -1 ENOENT (No such file or directory)
```

**Comment by bdimcheff** [comment]
I have the same crash if I try to print in a variety of apps, at least Firefox and Chrome (if you choose "Use system print dialog")

**Comment by abbradar** [comment]
@bdimcheff This should be fixed now that Firefox uses GTK3; what version of NixOS are you using?

**Comment by bdimcheff** [comment]
17.03 for my system, unstable as of yesterday for nixpkgs for my user, though I tried firefox from a fresh user that was using 17.03 packages and it still had the problem  (firefox-bin btw, I can't use firefox nightly because of #24295 currently)

**Comment by abbradar** [comment]
This might be an issue with `firefox-bin`, I'll check this and Chrome (Chrome or Chromium?) later.

**Comment by bdimcheff** [comment]
Both chrome and chromium stable, at ref efe12d6658 in nixpkgs.  Firefox (not-bin) also crashes when trying to print... I can't actually load a page successfully, but it'll still pull up the print dialog and crash immediately if I try to print the tab crashed page.

**Comment by abbradar** [comment]
@bdimcheff `firefox` should be fixed since 56ca347872ea8fd6e52d9cca19d42cc738e5569d and `firefox-bin` since 6df350ffc81e2a5891daa4962f68014378308c1c; native print dialog works for me in `chromium` and I'm not sure whether it needs `wrapGAppsHook`.

**Comment by abbradar** [comment]
Fixed by 55657286709b481a2c6e31d8447d006e1fdd25d1
